presto-main/src/main/java/io/prestosql/security/ViewAccessControl.java

```diff
@@ -42,7 +44,12 @@ public void checkCanSelectFromColumns(SecurityContext context, QualifiedObjectNa
         // In SQL, views are special in that they execute with permissions of the owner.
         // This means that the owner of the view is effectively granting permissions to the user running the query,
         // and thus must have the equivalent of the SQL standard "GRANT ... WITH GRANT OPTION".
-        delegate.checkCanCreateViewWithSelectFromColumns(context, tableName, columnNames);
+        try {
+            delegate.checkCanCreateViewWithSelectFromColumns(context, tableName, columnNames);
+        }
+        catch (AccessDeniedException e) {
```

Comment:
We should fix the callers instead of catching and rethrowing here

Reply:
@electrum In this case we clearly state that it's OWNER(delegate) who lacks privileges and it's possible to find why he lacks these privileges (as stacktrace is preserved). 

Otherwise functionality of AccessControlManager#checkCanCreateViewWithSelectFromColumns has to be duplicated (at least partially, e.g. AccessControlManager#checkCanAccessCatalog) to state that it is not current user, but view owner lacks privileges (at least to access catalog). Such duplication will produce a lot of code which does absolutely the same stuff.

Reply:
Maybe I'm not understanding the goal if this PR. Can you update the description to explain what "improve" means?

Reply:
done.